reyna/src/com/b2msolutions/reyna/services/StoreService.java

```diff
@@ -38,28 +40,56 @@ public static void setCellularDataBlackout(Context context, TimeRange range) {
         new Preferences(context).saveCellularDataBlackout(range);
     }
 
-	@Override
-	protected void onHandleIntent(Intent intent) {
+    public static void setStorageSizeLimit(Context context, long limit) {
```

Comment:
Should we not check if we have to shrink and vacuum the data-base whenever a new limit is set, that is smaller than the current limit.